x-pack/filebeat/input/httpjson/request.go

```diff
@@ -719,3 +847,15 @@ func drainBody(b io.ReadCloser) (r1 io.ReadCloser, err error) {
 
 	return io.NopCloser(&buf), nil
 }
+
+func closeChannels(intChannels []chan int, maybeMsgChannels []chan maybeMsg, httpResponseChannels []chan *http.Response) {
+	for _, c := range intChannels {
+		close(c)
+	}
+	for _, c := range maybeMsgChannels {
+		close(c)
+	}
+	for _, c := range httpResponseChannels {
+		close(c)
+	}
+}
```

Comment:
```suggestion
func closeChannels(chans ...chan *http.Response) {
	for _, c := range chans {
		close(c)
	}
}
```

Initially I had this as written below, but the non-`*http.Response` chans are never more than unity, so the genericisation is not required.

```
func closeChannels[E any](chans ...chan E) {
	for _, c := range chans {
		close(c)
	}
}
```
~~Then at the call sites you can `close(c)`, `closeChannels(c1, c2)` and  `closeChannels(c1, c2, c3)` (though there will need to be calls for each type of chan).~~

Reply:
Okay, I'll update it accordingly.